test/integration/test-extensions-loading.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createFixtureIframe,
+  expectBodyToBecomeVisible,
+} from '../../testing/iframe';
+
+function checkElementUpgrade(element) {
+  expect(element).to.have.class('-amp-element');
+  expect(element).to.have.class('-amp-layout-responsive');
+  expect(element).to.have.class('-amp-layout-size-defined');
+  expect(element).to.not.have.class('-amp-not-built');
+  expect(element).to.not.have.class('amp-not-built');
+  expect(element).to.not.have.class('amp-unresolved');
+  expect(element).to.not.have.class('-amp-unresolved');
+}
+
+describe.only('extensions loading multiple orders', () => {
+  let fixture;
+  it('one extension, extension loads first, all scripts in header', () => {
+    return createFixtureIframe(
+        'test/fixtures/script-load-extension-head-v0-head.html').then(f => {
+          fixture = f;
+          const ampFitText = fixture.doc.querySelectorAll('amp-fit-text');
+          expect(ampFitText).to.have.length(1);
+          return expectBodyToBecomeVisible(fixture.win);
```

Comment:
I think it would be more robust to wait for the `fixture.awaitEvent('amp:load:start', $count)` instead. Then you really know that AMP has proceeded to render the elements.